conans/test/functional/tools/test_files.py

```diff
@@ -116,6 +116,61 @@ def build(self):
     assert 'mypkg/1.0: Apply patch (security)' in str(client.out)
 
 
+def test_patch_real():
+    conanfile = textwrap.dedent("""
```

Comment:
Please @lasote review this new test:

- Seems to work well, I'd say still intuitive
- The only thing I slightly struggled is the new ``patch()`` function assuming it works on the ``conanfile.source_folder`` to apply patches to ``build()``. I needed to pass ``base_path=self.build_folder``, but that was unexpected I assumed it was operating on ``cwd``

Reply:
Indeed that was unexpected. I always though about patches in "source" but probably the line `root = os.path.join(conanfile.source_folder, base_path) if base_path else conanfile.source_folder` should be changed.